integration-tests/tests/src/tests/sync/user.ts

```diff
@@ -67,7 +67,11 @@ function removeExistingUsers(): void {
     if (this.app) {
       const users = this.app.allUsers;
       for (const userId of Object.keys(this.app.allUsers)) {
-        await this.app.removeUser(users[userId]);
+        if (users[userId].isLoggedIn) {
+          await this.app.deleteUser(users[userId]);
+        } else {
+          await this.app.removeUser(users[userId]);
+        }
```

Comment:
Suggesting this change as per [this thread](https://github.com/realm/realm-js/pull/6032#discussion_r1331456122).
```suggestion
        await this.app.removeUser(users[userId]);
```